packages/foundation-models/src/azure-openai/stream/line-decoder.ts

```diff
@@ -0,0 +1,105 @@
+type Bytes = string | ArrayBuffer | Uint8Array | Buffer | null | undefined;
+
+/**
+ * A re-implementation of httpx's `LineDecoder` in Python that handles incrementally
+ * reading lines from text.
+ *
+ * Https://github.com/encode/httpx/blob/920333ea98118e9cf617f246905d7b202510941c/httpx/_decoders.py#L258.
+ * @internal
+ */
+export class LineDecoder {
```

Comment:
[q] Wouldn't the `line-decoder`, `sse-stream` and `sse-decoder` be eventually used for Orchestration streaming as well? Can we move them to `core` package instead?

Reply:
This will be part of the orchestration streaming PR. As only then, we know how orchestration service behaves and maybe certain modifications will be needed.

Reply:
Hmm, looks like [they will](https://github.tools.sap/AI/llm-orchestration/blob/main/docs/adr/008-output-streaming.md) also use SSE.

But, since, we don't expose these types to users anyway, you are right, we can move them later too.